typing/typecore.ml

```diff
@@ -3179,6 +3191,21 @@ let check_recursive_class_bindings env ids exprs =
          raise(Error(expr.cl_loc, env, Illegal_class_expr)))
     exprs
 
+(* The "rest of the function" extends from the start of the first parameter
+   to the end of the overall function. The parser does not construct such
+   a location so we forge one for type errors.
+*)
+let loc_rest_of_function
```

Comment:
Should also be called in `type_function` as this is supposed to be a refactoring.